### PR TITLE
Fixed all tests that broke.

### PR DESCRIPTION
**Pull request by SandeepThokala**

### CODE REVIEW

**Comment by SandeepThokala** [comment]
It doesn't change our Python version requirement, because HexSE is developed under version 3.6.9 and Type Hints have were introduced as a new feature in version 3.5

**Comment by ArtPoon** [comment]
@SandeepThokala adding some unit tests to improve coverage, hold off on review until those changes have been commited to his branch

File: tests/test_run_simulation.py [hunk 1]
```diff
@@ -556,5 +565,132 @@ def test_user_spec_mu(self):
...
+class TestResolveAmbiguities(unittest.TestCase):
+
+    def setUp(self):
+        random.seed(4973)
+
+    def testBadInput(self):
+        seq = "SGDKLJFAHHH"
+        expected = "CGGGLJFAACA"
+        result = resolve_ambiguities(seq)
+        self.assertEqual(expected, result)
```
Comment (ArtPoon) [inline tests/test_run_simulation.py:577]: There should also be a check here that, after resolving ambiguities, the resulting sequence is rejected as containing illegal characters

File: tests/test_run_simulation.py [hunk 2]
```diff
@@ -327,28 +329,28 @@ def test_discretize_gamma(self):
         alpha = 1.25
         ncat = 4
         expected = np.array([0.09199853806558903, 0.27043066909631136, 0.5158061369385518, 1.1217646558655263])
-        result = discretize(alpha, ncat, dist=ss.gamma)
+        result = discretize(alpha, ncat, dist=ss.gamma, scale=0)
```
Comment (ArtPoon) [inline tests/test_run_simulation.py:332]: Setting `scale` to zero does not make sense - this  should not produce reasonable outputs!

**Comment by ArtPoon** [comment]
The problem with `scale=0` is here:
https://github.com/PoonLab/HexSE/blob/cdcafac9066dcc6e5aebf89c077e7f1f45ceaecf/hexse/discretize.py#L25-L26

This check assumes that `scale` is `None`, which we use to indicate that the gamma distribution is a one-parameter distribution (with mean constrained to 1) - however if we pass `scale=0` then the condition also evaluates to true and `scale` is reset to `1/alpha`.

We should replace this condition with `if scale is None:` and also need to document the behaviour of `scale` set to `None`.

**Pull request merged**